modules/nf-core/seurat/multiseq/templates/MULTIseq.R

```diff
@@ -0,0 +1,197 @@
+# Function to transform strings to the correct class
+convert_element <- function(x) {
+        if (is.character(x)) {
+    num_value <- suppressWarnings(as.numeric(x))
+    if (!is.na(num_value)) {
+        return(num_value)
+    }
+
+    # Try to convert to boolean
+    bool_value <- tolower(x)
+    if (bool_value %in% c("true", "t", "yes", "y", "1")) {
+        return(TRUE)
+    } else if (bool_value %in% c("false", "f", "no", "n", "0")) {
+        return(FALSE)
+    }
+        }
+        return(x)
+}
+process_list <- function(input_list) {
+        return(lapply(input_list, convert_element))
+}
+opt_args_transformed <- process_list(args_opt)
+opt <- process_list(opt)
+
+################################################
+################################################
+## Create Seurat object                       ##
+################################################
+################################################
+#### Seurat object creation following tutorial from: https://satijalab.org/seurat/articles/hashing_vignette
+library(Seurat)
+library(ggplot2)
+
+# Read data
+rna_mtx <- Read10X(opt\$rna_matrix)
+
+hto_mtx <- Read10X(opt\$hto_matrix)
+
+# Select cell barcodes detected by both RNA and HTO In the example datasets we have already
+# filtered the cells for you, but perform this step for clarity.
+joint_matrices <- intersect(colnames(rna_mtx), colnames(hto_mtx))
+
+# Subset RNA and HTO counts by joint cell barcodes
+rna_mtx <- rna_mtx[, joint_matrices]
+hto_mtx <- as.matrix(hto_mtx[, joint_matrices])
+
+# Confirm that the HTO have the correct names
+rownames(hto_mtx)
+
+# Setup Seurat object
+seurat_object <- CreateSeuratObject(counts = Matrix::Matrix(as.matrix(rna_mtx), sparse = T))
+# Normalize RNA data with log normalization
+seurat_object <- NormalizeData(seurat_object)
+# Find and scale variable features
+seurat_object <- FindVariableFeatures(seurat_object, selection.method = opt_args_transformed\$selection_method, nfeatures = opt_args_transformed\$nfeatures)
+seurat_object <- ScaleData(seurat_object, features = VariableFeatures(seurat_object))
+# Add HTO data as a new assay independent from RNA
+seurat_object[[opt_args_transformed\$assay]] <- CreateAssayObject(counts = hto_mtx)
+# Normalize HTO data, here we use centered log-ratio (CLR) transformation
+seurat_object <- NormalizeData(seurat_object, assay = opt_args_transformed\$assay, normalization.method = opt_args_transformed\$normalization_method)
+################################################
+## MULTIseq                       ##
+################################################
+demultiplex <- MULTIseqDemux(seurat_object,assay = opt_args_transformed\$assay, quantile = opt_args_transformed\$quantile, autoThresh = opt_args_transformed\$autoThresh, maxiter = opt_args_transformed\$maxiter, qrange = seq(from = opt_args_transformed\$qrange_from, to = opt_args_transformed\$qrange_to, by = opt_args_transformed\$qrange_by), verbose = opt_args_transformed\$verbose)
```

Comment:
Is the module intended predominantly as a wrapper around this function? Then maybe it should be called `seurat/MULTIseqDemux`?

Reply:
Ok, so should I create a new branch with the new name or simply rename the folder created in this branch?

Reply:
You can rename the folder, but you'll also need to change the process name at the top of the files (main.nf and meta.yml), and correspondingly inside the tests folder.